packages/api/lambdas/sf-event-sqs-to-db-records/index.js

```diff
@@ -59,12 +61,21 @@ const writeRecordsToDynamoDb = async (cumulusMessage) => {
 const writeRecords = async (cumulusMessage, knex) => {
   const executionArn = getMessageExecutionArn(cumulusMessage);
 
-  const collectionCumulusId = await getMessageCollectionCumulusId(cumulusMessage, knex);
-  const isExecutionRDSWriteEnabled = await shouldWriteExecutionToRDS(
+  const [
+    collectionCumulusId,
+    asyncOperationCumulusId,
+    parentExecutionCumulusId,
+  ] = await Promise.all([
+    getMessageCollectionCumulusId(cumulusMessage, knex),
```

Comment:
This was refactored because I need the cumulus ID for the async operation or parent execution (if they exist) to be passed to the execution write logic on lines 91-93

Reply:
This seems :100: code wise.     I'm wondering if there's a more optimized way of getting these connection/query wise, but I suppose for the moment that's YAGNI.   Something possibly to stash when we get to performance tuning though.